### ISSUE TITLE
Grid docs: explain sorter aria-label generation

### ISSUE THREAD

**Issue by rolfsmeds** [vaadin/docs#1833]
`aria-label` attributes are autogenerated for Grid column sorters based on the text content in the sorter, prefixed by `"Sort by "`, so e.g. a column with the header text "Surname" would get `aria-label="Sort by Surname"`. Currently [this does not work](https://github.com/vaadin/web-components/issues/3470#issuecomment-1288639127) with column headers with `ComponentRenderer`s.

These details should be mentioned in the Grid documentation (e.g. under Sorting section, or in a separate Accessibility section).
